HDF5 error on main

We've started getting an error on `main` ([link](https://github.com/zarr-developers/VirtualiZarr/actions/runs/9559364935/job/26349624374)):

```
FAILED virtualizarr/tests/test_xarray.py::TestLoadVirtualDataset::test_loadable_variables - RuntimeError: Unspecified error in H5DSget_num_scales (return value <0)
```

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -26,18 +26,21 @@ jobs:
     steps:
       - uses: actions/checkout@v4
 
-      - name: Setup Python
-        id: setup-python
-        uses: actions/setup-python@v5
+      - name: Setup micromamba
+        uses: mamba-org/setup-micromamba@v1
         with:
-          python-version: ${{ matrix.python-version }}
-          cache: pip
-          cache-dependency-path: pyproject.toml
+          environment-file: ci/environment.yml
+          cache-environment: true
+          create-args: >-
+            python=${{matrix.python-version}}
 
       - name: Install virtualizarr
         run: |
-           python -m pip install -e  ".[test]"
-
+           python -m pip install -e . --no-deps
+      - name: Conda list information
+        run: |
+          conda env list
+          conda list
 
       - name: Running Tests
         run: |

--- a/ci/environment.yml
+++ b/ci/environment.yml
@@ -1,0 +1,28 @@
+name: virtualizarr-tests
+channels:
+  - conda-forge
+  - nodefaults
+dependencies:
+  - h5netcdf
+  - h5py
+  - hdf5
+  - netcdf4
+  - xarray>=2024.5.0
+  - kerchunk>=0.2.5
+  - pydantic
+  - numpy
+  - ujson
+  - packaging
+  - universal_pathlib
+  # Testing
+  - codecov
+  - pre-commit
+  - ruff
+  - pytest-mypy
+  - pytest-cov
+  - pytest
+  - pooch
+  - scipy
+  - fsspec
+  - s3fs
+  - fastparquet

